spark-etl/src/main/scala/geotrellis/spark/etl/config/EtlConf.scala

```diff
@@ -0,0 +1,134 @@
+    is.close(); fs.close(); json
+  }
+
+  def nextOption(map: Map[Symbol, String], list: Seq[String]): Map[Symbol, String] =
+    list.toList match {
+      case Nil => map
+      case "--input" :: value :: tail =>
+        nextOption(map ++ Map('input -> value), tail)
+      case "--backend-profiles" :: value :: tail =>
+        nextOption(map ++ Map('backendProfiles -> value), tail)
+      case "--output" :: value :: tail =>
+        nextOption(map ++ Map('output -> value), tail)
+      case "--help" :: tail => {
+        println(help)
+        sys.exit(1)
+      }
+      case option :: tail => {
+        println(s"Unknown option ${option}")
+        println(help)
+        sys.exit(1)
+      }
+    }
+
+  def parse(args: Seq[String])(implicit sc: SparkContext) =
+    nextOption(Map(), args).map { case (key, value) => key -> getJson(value, sc.hadoopConfiguration) }
+
+  def apply(args: Seq[String])(implicit sc: SparkContext) = {
+    val m = parse(args)
+
+    if(m.keySet != requiredFields) {
+      println(s"missing required field(s): ${(requiredFields -- m.keySet).mkString(", ")}, use --help command to get additional information about input options.")
+      sys.exit(1)
+    }
+
+    val(backendProfiles, input, output) = (m('backendProfiles), m('input), m('output))
+
+    val inputValidation           = inputSchema.validate(JsonLoader.fromString(input), true)
+    val backendProfilesValidation = backendProfilesSchema.validate(JsonLoader.fromString(backendProfiles), true)
+    val outputValidation          = outputSchema.validate(JsonLoader.fromString(output), true)
+
+    if(!inputValidation.isSuccess || !backendProfilesValidation.isSuccess || !outputValidation.isSuccess) {
+      if(!inputValidation.isSuccess) {
+        println("input validation error:")
+        println(inputValidation)
+      }
+      if(!backendProfilesValidation.isSuccess) {
+        println("backendProfiles validation error:")
+        println(backendProfilesValidation)
+      }
+      if(!outputValidation.isSuccess) {
+        println("output validation error:")
+        println(outputValidation)
+      }
+      sys.exit(1)
+    }
+
+    val backendProfilesParsed =
+      backendProfiles.parseJson.asJsObject.getFields("backend-profiles") match { case Seq(bp: JsArray) =>
+        bp.elements.map { js: JsValue =>
+          js.asJsObject.getFields("name", "type") match {
+            case Seq(JsString(n), JsString(t)) => n -> (BackendType.fromString(t) match {
+              case HadoopType    => js.convertTo[HadoopProfile]
+              case S3Type        => js.convertTo[S3Profile]
+              case AccumuloType  => js.convertTo[AccumuloProfile]
+              case CassandraType => js.convertTo[CassandraProfile]
+            })
+          }
+        }.toMap
+      }
+
+    val inputsParsed = input.parseJson.convertTo[List[Input]]
+
+    val outputParsed = output.parseJson.convertTo[Output]
+
+    inputsParsed.map { inputParsed =>
```

Comment:
This is basically trying to do a lookup from profile name to profile instance. At this point you have parsed the profiles already. It would be just as easy to give that as a constructor to `JsonFormat[Inout]` instance defined here, which would be used for parsing. This would also allow you to upgrade `Backend` to have the actual profile.